### PR TITLE
query and update animals table

### PR DESCRIPTION
**Pull request by RileyManda**
I  have completed insertions, queries and transactions|aggregates on the Vet clinic project and updated the schema according to the stipulated [requirements ](https://github.com/microverseinc/curriculum-databases/blob/main/db-structure/vet_clinic_query_update_tables.md): 🌱 

**Data insertion**

- [x] Charmander, born on Feb 8th, 2020, currently weighs -11kg. He is not neutered and he has never tried to escape.
- [x] Plantmon, born on Nov 15th, 2021, and currently weighs -5.7kg. She is neutered and she has tried to escape 2 times.
- [x] Squirtle, born on Apr 2nd, 1993, and currently weighs -12.13kg. He was not neutered and he has tried to escape 3 times.
- [x] Angemon, born on Jun 12th, 2005, and currently weighs -45kg. He is neutered and he has tried to escape once.
- [x] Boarmon, born on Jun 7th, 2005, and currently weighs 20.4kg. He is neutered and he has tried to escape 7 times.
- [x] Blossom, born on Oct 13th, 1998, and currently weighs 17kg. She is neutered and she has tried to escape 3 times.
- [x] Ditto , born on May 14th, 2022, and currently weighs 22kg. He is neutered and he has tried to escape 4 times.

### Transactions

- [x]  Transaction[1]:

    - **Updated animals table by setting the species column to unspecified**

Screen Shot 2023-08-21 at 3 40 05 PM

    - Roll back transaction[1]

    
Screen Shot 2023-08-21 at 3 42 21 PM

- [x] Transaction[2]: 👇 

- Update the animals table by setting the species column to digimon for all animals that have a name ending in mon.

Screen Shot 2023-08-21 at 3 51 15 PM

- [x] Transaction[3]: 👇 

- Set the species column to pokemon for all animals that don't have species already set.

Screen Shot 2023-08-21 at 3 54 26 PM

- [x] Transaction[4]: 👇 
- delete all records in the animals table, then roll back the transaction.

Screen Shot 2023-08-21 at 3 57 51 PM

       - Roll back transaction[4] 👇 

Screen Shot 2023-08-21 at 3 59 46 PM

- [x] Transaction[5]: Delete all animals born after Jan 1st, 2022: 👇 

Screen Shot 2023-08-21 at 4 08 54 PM

- Create a savepoint for the transaction.👇 

Screen Shot 2023-08-21 at 4 17 22 PM

- [x] Transaction[6]: Update all animals' weight to be their weight multiplied by -1. 👇 

Screen Shot 2023-08-21 at 4 21 10 PM

- [x] Transaction[7]: Update all animals' weights that are negative to be their weight multiplied by -1.

Screen Shot 2023-08-21 at 4 47 52 PM

### Queries

- Count number of animals

Screen Shot 2023-08-21 at 4 53 29 PM

- Count the number of animals that have never escaped:

Screen Shot 2023-08-21 at 4 54 28 PM

- Average weight of the animals:

Screen Shot 2023-08-21 at 4 56 03 PM

- Escape attempts

Screen Shot 2023-08-21 at 5 00 15 PM

- Minimum and maximum weight of each type of animal:

Screen Shot 2023-08-21 at 5 04 53 PM

- Average number of escape attempts per animal type-born between 1990 and 2000?
Screen Shot 2023-08-21 at 5 07 34 PM

Please kindly review my implementation and suggest any changes.

Thank you 🙏

### CODE REVIEW

File: queries.sql [hunk 1]
```diff
@@ -8,4 +8,87 @@ SELECT date_of_birth FROM animals WHERE name IN ('Agumon', 'Pikachu');
...
+
+-- minimum and maximum weight of each type of animal
+SELECT species, MIN(weight_kg) AS min_weight, MAX(weight_kg) AS max_weight
+FROM animals
+GROUP BY species;
+
+--Average number of escape attempts per animal type-born between 1990 and 2000?
+SELECT species, AVG(escape_attempts) AS avg_escape_attempts
+FROM animals
+WHERE date_of_birth BETWEEN '1990-01-01' AND '2000-12-31'
```
Comment (Olamarx) [inline queries.sql]: - [ ] When a query is less than or equal to, the logic requires the date to be backdated, in this case, it should be from 1989-12-31 and 2000-12-30. Do you understand? If you don't, I can explain further, just tag me.

**Review by Olamarx — CHANGES_REQUESTED** [review]
Hi @RileyManda,

Good job so far, indeed you have spent time on this project!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights
- Queries, data, and schema correct ✔️
- Readme file professional ✔️
- Good commit messages ✔️

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: queries.sql (hunk 1 above)
Comment (RileyManda) [reply queries.sql]: Hi @Olamarx 

Thank you for the review and feedback.
Yes,I understand.Thank you for suggesting this change.
I will work on the update and re-submit for review. 🙏

**Comment by RileyManda** [comment]
I have update my implementation with changes suggested by the reviewers:

Implementations made: 🌱 

- [x] Update date to back0dated date for Average number of escape attempts query:

Screen Shot 2023-08-21 at 5 54 32 PM

Please kindly review my implementation and suggest any changes.

Thank you 🙏

**Review by AbelG101 — APPROVED** [review]
# Hi @RileyManda 👋, 

**Your project is complete!  There is nothing else to say other than... it's time to merge it. 
Congratulations! 🎉 🥳🥇**

## Highlights
- Date is properly updated to match given question✔
- Species column is properly added ✔
- Proper transaction management ✔
- Correct use of Git Flow ✔
- Descriptive PR title with a great screenshot summary ✔

## Optional suggestions
- N/A

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**